examples/orange/Dockerfile

```diff
@@ -0,0 +1,86 @@
+# run `make build` to build local/elemental-toolkit image
+ARG TOOLKIT_REPO
+ARG VERSION
+FROM ${TOOLKIT_REPO}:${VERSION} as TOOLKIT
+
+# OS base image of our choice
+FROM ubuntu:22.04 as OS
+ARG REPO
+ARG VERSION
+ENV VERSION=${VERSION}
+
+# install kernel, systemd, dracut, grub2 and other required tools
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  linux-generic \
+  dmsetup \
+  dracut-core \
+  dracut-network \
+  dracut-live \
+  dracut-squash \
+  grub2-common \
+  grub-efi-amd64 \
+  shim \
+  shim-signed \
+  haveged \
+  systemd \
+  systemd-sysv \
+  systemd-timesyncd \
+  openssh-server \
+  openssh-client \
+  tzdata \
+  parted \
+  e2fsprogs \
+  dosfstools \
+  mtools \
+  xorriso \
+  findutils \
+  gdisk \
+  rsync \
+  squashfs-tools \
+  lvm2 \
+  vim \
+  less \
+  sudo \
+  ca-certificates \
+  curl \
+  iproute2 \
+  dbus-daemon \
+  patch \
+  netplan.io \
+  xz-utils
+
+# Create non FHS paths
+RUN mkdir -p /oem /system
+
+# Just add the elemental cli
+COPY --from=TOOLKIT /usr/bin/elemental /usr/bin/elemental
+
+# Enable essential services
+RUN systemctl enable systemd-networkd.service && \
+    systemctl enable systemd-resolved.service
+
+# Enable /tmp to be on tmpfs
+RUN cp /usr/share/systemd/tmp.mount /etc/systemd/system
+
+# Generate initrd with required elemental services
+RUN elemental --debug init -f
+
+# Update os-release file with some metadata
+RUN echo IMAGE_REPO=\"${REPO}\"         >> /etc/os-release && \
+    echo IMAGE_TAG=\"${VERSION}\"           >> /etc/os-release && \
+    echo IMAGE=\"${REPO}:${VERSION}\" >> /etc/os-release && \
+    echo TIMESTAMP="`date +'%Y%m%d%H%M%S'`" >> /etc/os-release
+RUN echo GRUB_ENTRY_NAME=\"Elemental\" >> /etc/os-release
+
+# Adding specific network configuration based on netplan
+ADD 05_network.yaml /system/oem/05_network.yaml
+
+# Arrange bootloader binaries into /boot/efi/EFI/elemental 
+# this way elemental installer can easily fetch them
+RUN mkdir -p /boot/efi/EFI/elemental && \
+    cp /usr/lib/grub/x86_64-efi-signed/gcdx64.efi.signed /boot/efi/EFI/elemental/grubx64.efi && \
+    cp /usr/lib/shim/shimx64.efi.signed.latest /boot/efi/EFI/elemental/shimx64.efi && \
+    cp /usr/lib/shim/mmx64.efi /boot/efi/EFI/elemental/mmx64.efi
```

Comment:
@frelon in this section I mostly exercise this concept of a custom path (`/boot/efi/EFI/elemental`) to arrange the bootloader binaries in a way that elemental installer will find them. However I not convinced about this path `/boot/efi/EFI/elemental`, as `/boot/efi` is usually the mountpoint of EFI partition in many distributions, so I am wondering if shouldn't we use something like `/etc/elemental/bootloader` or something else. Do you have any better idea? The more I think about `/boot/efi/EFI/elemental` the more I dislike it.


Probably a structure like like:


```
/etc/elemental/config.yaml
/etc/elemental/config.d/extra_config.yaml
/etc/elemental/bootloader/grub.cfg
/etc/elemental/bootloader/bootargs.cfg
/etc/elemental/bootloader/shim.efi
/etc/elemental/bootloader/grub.efi
/etc/elemental/bootloader/mmx64.efi
...
```

Reply:
Yep, I agree that `/boot/efi` is not ideal! It seems at least tumbleweed puts some bootloader stuff in `/usr/lib/bootloader`.. Maybe we could use something like `/usr/lib/bootloader/elemental` for efi apps and bootloader config?

Reply:
I'm not opposed to using `/etc/elemental/bootloader` either!

Reply:
interesting, just checked the `/usr/lib/bootloader` is owned by the package perl-Bootloader (perl utilities for various bootloaders). I don't want to get mixed with this. But you got a point, probably we should use something like `/usr/lib/elemental/bootloader`. My resistance of `/etc/elemental` is that it feels odd adding binaries in there, `/etc` should include only configuration files.

What about:

```
/etc/elemental/config.yaml
/etc/elemental/config.d/extra_config.yaml
/etc/elemental/bootloader/grub.cfg
/etc/elemental/bootloader/bootargs.cfg
/usr/lib/elemental/bootloader/shim.efi
/usr/lib/elemental/bootloader/grub.efi
/usr/lib/elemental/bootloader/mmx64.efi
```

Anyway we can run this change on a follow up PR.

Reply:
Looks reasonable to me!

Reply:
Finally used `/usr/lib/elemental/bootloader` after rebasing over the riscv64 support.